### DISCUSSION TITLE
version 2 b3 - pcm_alaw codec not supported

### DISCUSSION THREAD

**Discussion by Tomiol80** [roflcoopter/viseron#396] [General]
I have two Tapo C100 cameras and I cannot get recording in working state.

I get error messages:
viseron    | [2022-11-24 21:29:56] [ERROR   ] [viseron.components.ffmpeg.stream.tallikamppa] - [mp4 @ 0x558bb1
4199c0] Could not find tag for codec pcm_alaw in stream #1, codec not currently supported in container
viseron    | [2022-11-24 21:29:56] [ERROR   ] [viseron.components.ffmpeg.stream.tallikamppa] - Could not write
 header for output file #0 (incorrect codec parameters ?): Invalid argument

Audio cannot be disabled on those cameras.

Is it possible to add the codec to Viseron?

**Comment by roflcoopter** [comment]
It is not a limitation in Viseron, but in FFmpeg which cant handle `pcm_alaw` in an `mp4` container.

To get around this you can set `extension: mkv` under `recorder` something like this:

```yaml
ffmpeg:
  camera:
    porch:
      name: porch
      host: blabla
      ...
      recorder:
        extension: mkv
